apps/system/js/bluetooth_v2.js

```diff
@@ -0,0 +1,438 @@
+'use strict';
+/**
+ * Bluetooth2 is compatible with Bluetooth APIv2 and used to enable/
+ * disable bluetooth hardware, get bluetooth adapter, and check target
+ * bluetooth profile is connected.
+ *
+ * Bluetooth api v2 is the two level structure.
+ * We use _bluetoothManagerHandler to handle BluetoothManager defaultAdapter
+ * change event and _bluetoothAdapterHandler to handle status change event.
+ *
+ */
+/* global SettingsListener, Service */
+/* exported Bluetooth2 */
+(function(exports) {
+
+var Bluetooth = function() {};
+
+Bluetooth.prototype = {
+  name: 'Bluetooth',
+  /**
+   * Keep a global connected property.
+   *
+   * @public
+   */
+  connected: false,
+
+  /**
+   * Debug message.
+   *
+   * @type {Boolean} turn on/off the console log
+   */
+  onDebug: false,
+
+  /**
+   * Store a reference of the default adapter.
+   *
+   * @private
+   */
+  _adapter: null,
+
+  /**
+   * Hold instance of bluetooth attribute change handler.
+   *
+   * @private
+   */
+  _defaultAdapterChangeHandler: null,
+
+  /**
+   * Hold instance of bluetooth adapter state change handler.
+   *
+   * @private
+   */
+  _stateChangeHandler: null,
+
+  /**
+   * Hold instance of bluetooth attribute change handler in
+   * getAdapter when the default adapter is not available.
+   *
+   * @private
+   */
+  _promiseAdapterChangeHandler: null,
+
+  /**
+   * Hold instance of the getAdapter promise
+   * when the default adapter is not available.
+   */
+  _cacheGetAdapterPromise: null,
+
+  /**
+   * Store a reference of the Bluetooth API.
```

Comment:
It's not clear for me. The document is not figuring out what the API is stored here.
How about: `Store a reference  for mozBluetooth.`